net/tlsdial/tlsdial.go

```diff
@@ -79,18 +92,21 @@ func Config(host string, base *tls.Config) *tls.Config {
 		if debug() {
 			log.Printf("tlsdial(sys %q): %v", host, errSys)
 		}
-		if errSys == nil {
-			return nil
-		}
 
-		// If that failed, because the system's CA roots are old
-		// or broken, fall back to trying LetsEncrypt at least.
+		// Always verify with our baked-in Let's Encrypt certificate,
+		// so we can log an informational message. This is useful for
+		// detecting SSL MiTM.
 		opts.Roots = bakedInRoots()
-		_, err := cs.PeerCertificates[0].Verify(opts)
+		_, bakedErr := cs.PeerCertificates[0].Verify(opts)
 		if debug() {
-			log.Printf("tlsdial(bake %q): %v", host, err)
+			log.Printf("tlsdial(bake %q): %v", host, bakedErr)
+		} else if bakedErr != nil {
+			log.Printf("tlsdial: server cert for %q is not a Let's Encrypt cert", host)
```

Comment:
this'll be spammy for e.g. all the headscale users in China or anybody running a custom derper with a non-LE cert

Reply:
Good point; let me rate-limit this.

Reply:
We now log only once per `host` per process invocation; PTAL? I could switch this to "every hour/day/etc." instead if you'd rather.